pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -599,18 +600,26 @@ func (s *awsSdkEC2) DescribeSecurityGroups(request *ec2.DescribeSecurityGroupsIn
 }
 
 func (s *awsSdkEC2) AttachVolume(request *ec2.AttachVolumeInput) (*ec2.VolumeAttachment, error) {
-	return s.ec2.AttachVolume(request)
+	requestTime := time.Now()
+	resp, err := s.ec2.AttachVolume(request)
+	timeTaken := time.Since(requestTime).Seconds()
+	recordAwsMetric("aws_attach_volume", timeTaken, "")
```

Comment:
This will record duration for successful and failed calls in the same metric. I'd prefer only recording them for succesfull  requests, but I know some people have different preferences.